sql/binlogreplication/binlog_replication.go

```diff
@@ -62,6 +64,31 @@ type BinlogReplicaController interface {
 	ResetReplica(ctx *sql.Context, resetAll bool) error
 }
 
+// BinlogPrimaryController allows GMS to coordinate with a binlog primary server. Providers built on go-mysql-server may
+// optionally implement this interface and use it when constructing a SQL engine in order to receive callbacks when
+// replication statements for a primary server are received (e.g. SHOW BINARY LOG STATUS) or when MySQL protocol
+// commands related to replication are received (e.g. COM_REGISTER_REPLICA).
+type BinlogPrimaryController interface {
+	// RegisterReplica tells the binlog primary controller to register a new replica on connection |c| with the
+	// primary server. |replicaHost| and |replicaPort| specify where the replica can be accessed, and are returned
+	// from the SHOW REPLICAS statement.
+	RegisterReplica(ctx *sql.Context, c *mysql.Conn, replicaHost string, replicaPort uint16) error
+
+	// BinlogDumpGtid tells the binlog primary controller to start streaming binlog events to the replica over
+	// connection |c|. |gtidSet| specifies the point at which to start replication, or if it is nil, then it indicates
+	// the complete history of all transactions should be sent over the connection.
+	BinlogDumpGtid(ctx *sql.Context, c *mysql.Conn, gtidSet mysql.GTIDSet) error
+
+	// ListReplicas is called when the SHOW REPLICAS statement is executed.
```

Comment:
Any notes on the behavior here? Is this just a callback, since there's no output?

Same comment on the method below this

Reply:
I'll add some more explanation. I mostly wanted to get those stubbed out in the interface since I know GMS will need to call into the integrator's code for those, but they will still need to change to return some data to GMS.